src/WorkerChannel.ts

```diff
@@ -61,4 +71,13 @@ export class WorkerChannel {
             after(context);
         }
     }
+
+    public async initAppDir(dir: string) {
+        this.functionAppDir = dir;
+        try {
+            this.packageJson = await readJson(path.join(this.functionAppDir, 'package.json'));

```

Comment:
it looks like we will have an additional file system call during initialization and for consumption/EP plans we use azure storage file share which is pretty slow. I am wondering about cold start impact here.

Reply:
I don't think its a huge concern here. For example in https://github.com/Azure/azure-functions-nodejs-worker/issues/449, my understanding is only the last two entries were problematic and this file read should be like the first 9, not the last 2. Specifically, I expect the path to be like `C:\home\site\wwwroot\package.json`

We already do a file read for every function script file [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/FunctionLoader.ts#L41), and since users can have many functions I certainly hope one file read isn't a big performance impact.